homeassistant/components/digital_ocean/exceptions.py

```diff
@@ -0,0 +1,16 @@
+"""Store all custom exceptions for this integration."""
+
+
+from homeassistant.exceptions import HomeAssistantError
+
+
+class DomainRecordsNotFound(HomeAssistantError):
+    """Error used to indicate no domain records are found in Digital Ocean."""
+
+
+class UpdateThrottled(HomeAssistantError):
+    """Error used to indicate too frequent updates."""
+
+
+class DomainRecordAlreadySet(HomeAssistantError):
+    """Exception used to indicate update was skipped."""
```

Comment:
> Integrations should be updated and raise ServiceValidationError instead of ValueError when the user did something wrong, and HomeAssistantError when it's not the user's fault. [Read more](https://developers.home-assistant.io/docs/core/platform/raising_exceptions).

None of these are caused by the user. They are all raised by the outcome of the update call.
IMO these should remain `HomeAssistantError`

Reply:
Thanks @jbouwh 👍
With this in mind, can you please check again the correct base class @phrfpeixoto ?

Reply:
Great 👍  I've set the PR to draft status, please mark it as ready for review when you're done.

Reply:
@emontnemery Updated. Thank you for the input.